internal/icinga2/client_api.go

```diff
@@ -427,26 +438,55 @@ func (client *Client) listenEventStream() error {
 		)
 		switch respT := resp.(type) {
 		case *StateChange:
-			// Only process HARD states
+			// Ignore any state changes while in downtime
+			if respT.DowntimeDepth > 0 {
+				client.Logger.Debugf("Skipping state change, checkable is in downtime, %#v", respT)
+				continue
+			}
+
+			// Only process HARD states.
+			// The checkable might also be in the flapping state, but Icinga 2 doesn't expose "is_flapping" to the
+			// event streams for state change. So, the only thing we can do here is to ignore soft state changes.
```

Comment:
I'm fully aware that you added this based on [my earlier suggestion](https://github.com/Icinga/icinga-notifications/pull/146#issuecomment-2079014352):

> Another thing that just came to my mind where I don't immediately know what to do here: currently, this PR will happily send further state notifications while in downtime/flapping. At the moment, we don't have a mute incident functionality (which these events should probably trigger), so it might be an idea to ignore state events if in downtime/flapping, similar to how soft states are ignored.

However, after having played with this implementation, I noticed that this introduces quite some quirks due simply discarding a state change: after a downtime ended, the incident may still have the wrong severity, or it might even not be opened/closed even though it should.

So yes, while sending state notifications in a downtime sounds obviously undesirable, it might actually be the lesser evil until we properly tackle incident muting (#190). At least "downtimes are currently ignored when sending state notifications" is easier to understand than "in these circumstances (that would have to be explained additionally), an incident may go completely out of sync with what's happening in the source".

Reply:
Does this also include the catch-up phase or just this one? I am asking this because this is not the only place where the state changes are being ignored.

Reply:
In both places, they should stay consistent. (That's also another reason for postponing it to proper incident muting as currently these are inconsistent with flapping due due missing information in the event stream).